indico/modules/events/registration/controllers/management/regforms.py

```diff
@@ -252,6 +255,37 @@ def _process(self):
                                                     regform=self.regform)
 
 
+class RHRegistrationFormNotificationPreview(RHManageRegFormBase):
+    """Preview registration emails."""
+
+    @no_autoflush
+    @use_kwargs({
+        'message': fields.String(load_default=''),
+        'state': fields.String(load_default=RegistrationState.pending.name)
+    })
+    def _process(self, message, state):
+        self.commit = False
+        state = RegistrationState.get(state)
```

Comment:
If you use `'state': fields.Enum(RegistrationState, required=True),` above, you can get rid of this (also no need for a default value since you always expect to provide the state anyway).

You can also add `validate=validate.OneOf([...]))` to ensure only the expected states will be accepted.

Reply:
Done!